### DISCUSSION TITLE
Welcome to ecommerce-modular-monolith Discussions!

### DISCUSSION THREAD

**Discussion by mehdihadeli** [mehdihadeli/food-delivery-modular-monolith#17] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by FWDesperado** [comment]
Hi, I'm Des, @FWDesperado on several platforms. Did the ecommerce-modular-monolith project move? 
I am trying to learn as much as possible on MongoDB integration in ASP.NET Core minimal API apps as quickly as I can. The Mongo BuildingBlock project looks awesome. It's the type of code example that I've been looking for.

**Comment by FWDesperado** [comment]
Do you have an example of how you would call `AddMongoDbContext` when setting up DI? I am trying to get this working, but VS keeps telling me that it can not infer the parameter types. I'm sure that I'm missing something simple, but dependency injection example would help out a lot. I am able to call `builder.services.AddTransient<IMongoDbContext, MongoDbContext>()` and setup the options in the constructor, but that doesn't seem very clean.

**Reply by mehdihadeli** [reply]
After importing `BuildingBlocks.Persistence.Mongo` namespace you should able to add `AddMongoDbContext` to your service collections without any problem. could you check it out? also you should copy [BuildingBlocks.Persistence.Mongo](https://github.com/mehdihadeli/food-delivery-modular-monolith/tree/main/src/BuildingBlocks/BuildingBlocks.Persistence.Mongo) project beside your project and add this project to your solution file and after that you have access to it.

**Comment by mehdihadeli** [comment]
Hi,
Thanks for your feedback.
I'm refactor the project a little bit and It will be ready about next  2 week, and it is a complete example and you can just follow my example.
